### PR TITLE
Automatic language detection (W.I.P.)

### PR DESCRIPTION
**Pull request by ru-ka**

### CODE REVIEW

File: src/glowCookies.js [hunk 1]
```diff
@@ -235,55 +235,76 @@ class GlowCookies {
...
+    }
   }
 }
 
 class LanguagesGC {
-  constructor(code) {
+  constructor(code, defaultLang) {
     this.init()
-    let lang = this.arrLang[code] || this.arrLang['en']
+    var lang = null;
```
Comment (manucaralmo) [inline src/glowCookies.js:290]: Let's use 'let' to maintain consistency with the rest of the project.
```suggestion
    let lang = null;
```

File: src/glowCookies.js [hunk 2]
```diff
@@ -235,55 +235,76 @@ class GlowCookies {
...
+    if (code == "auto") {
+      var languages = [].concat(window.navigator.languages || [window.navigator.language || window.navigator.userLanguage]);
```
Comment (manucaralmo) [inline src/glowCookies.js:292]: Same as before
```suggestion
      let languages = [].concat(window.navigator.languages || [window.navigator.language || window.navigator.userLanguage]);
```

File: src/glowCookies.js [hunk 3]
```diff
@@ -235,55 +235,76 @@ class GlowCookies {
...
+      if (typeof defaultLang == "string")
+        languages.push(defaultLang);
+      languages.push('en');
+      for (let i = 0; i < languages.length; i++) {
+        // To-do: BCP47 to ISO-639 conversion
```
Comment (manucaralmo) [inline src/glowCookies.js:297]: Is it necessary to launch the functionality?

**Review by manucaralmo — COMMENTED** [review]
Great job! I think your contributions are a very good idea.

File: src/glowCookies.js (hunk 3 above)
Comment (ru-ka) [reply src/glowCookies.js:297]: It works even without, but it's preferable to implement. For example: in the languages array we can get things like `en-US`, `it-CH`, `pt-BR` and also `art-jbo` (wich is the same as `jbo`).

**Pull request closed without merging**